pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/download/DownloadParticipantListServiceTest.java

```diff
@@ -0,0 +1,174 @@
+                assertEquals("Waiting on GP", kitRequestShippingMap.get("sampleQueue"));
+            } else if (kitRequestShippingMap.get("ddpLabel").equals("shippedKit")) {
+                assertEquals("Shipped", kitRequestShippingMap.get("sampleQueue"));
+            } else {
+                Assert.fail();
+            }
+        });
+    }
+
+    private QueryParamsMap buildMockQueryParams(boolean onlyMostRecent, boolean humanReadable, String fileFormat) {
+        QueryParamsMap mainQueryParamsMap = mock(QueryParamsMap.class);
+
+        // Mocking the nested QueryParamsMap
+        QueryParamsMap nestedParentQueryParamsMap = mock(QueryParamsMap.class);
+        when(mainQueryParamsMap.get("parent")).thenReturn(nestedParentQueryParamsMap);
+        when(nestedParentQueryParamsMap.value()).thenReturn("participantList");
+
+        QueryParamsMap nestedOnlyMostRecentQueryParamsMap = mock(QueryParamsMap.class);
+        when(mainQueryParamsMap.get("onlyMostRecent")).thenReturn(nestedOnlyMostRecentQueryParamsMap);
+        when(nestedOnlyMostRecentQueryParamsMap.value()).thenReturn(String.valueOf(onlyMostRecent));
+
+        QueryParamsMap nestedRealmQueryParamsMap = mock(QueryParamsMap.class);
+        when(mainQueryParamsMap.get("realm")).thenReturn(nestedRealmQueryParamsMap);
+        when(nestedRealmQueryParamsMap.value()).thenReturn(ddpInstanceDto.getInstanceName());
+
+        QueryParamsMap nestedHumanReadableQueryParamsMap = mock(QueryParamsMap.class);
+        when(mainQueryParamsMap.get("humanReadable")).thenReturn(nestedHumanReadableQueryParamsMap);
+        when(nestedHumanReadableQueryParamsMap.value()).thenReturn(String.valueOf(humanReadable));
+
+        QueryParamsMap nestedUserIdQueryParamsMap = mock(QueryParamsMap.class);
+        when(mainQueryParamsMap.get("userId")).thenReturn(nestedUserIdQueryParamsMap);
+        when(nestedUserIdQueryParamsMap.value()).thenReturn("6");
+
+        QueryParamsMap nestedFileFormatQueryParamsMap = mock(QueryParamsMap.class);
+        when(mainQueryParamsMap.get("fileFormat")).thenReturn(nestedFileFormatQueryParamsMap);
+        when(nestedFileFormatQueryParamsMap.value()).thenReturn(fileFormat);
+
+        QueryParamsMap nestedFiltersQueryParamsMap = mock(QueryParamsMap.class);
+        when(mainQueryParamsMap.get("filters")).thenReturn(nestedFileFormatQueryParamsMap);
+        when(nestedFileFormatQueryParamsMap.value()).thenReturn("[]");
+
+        return mainQueryParamsMap;
+    }
+
+    private ManualFilterParticipantList getFilterFromFile(String fileName){
+        try {
+            String filterJson = TestUtil.readFile(fileName);
+            filterJson = filterJson.replace(":shortId", shortId);
+            ManualFilterParticipantList filterable = new Gson().fromJson(filterJson, ManualFilterParticipantList.class);
+            return filterable;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
Assert.fail or a more specific exception with something above catching it would work better.